src/compiler/utilities.ts

```diff
@@ -2414,6 +2414,27 @@ namespace ts {
         return output;
     }
 
+    export const stringify: (value: any) => string = JSON && JSON.stringify ? JSON.stringify : function stringify(value: any): string {

```

Comment:
True, but this is generally only for simple use cases (e.g. serializing source maps) where there are no circular references.